### PR TITLE
Perception : add customized tensorRT layer

### PR DESCRIPTION
**Pull request by KaWaiTsoiBaidu**

### CODE REVIEW

File: modules/perception/inference/tensorrt/plugins/BUILD [hunk 1]
```diff
@@ -66,6 +67,21 @@ cuda_library(
...
+    hdrs = [
+        "leakyReLU_plugin.h",
+    ],
+    deps = [
+        "//modules/perception/inference/tensorrt:rt_common",
+        "@cuda",
+        "@eigen",
+        "@tensorrt",
+    ],
+)
```
Comment (xiaoxq) [inline modules/perception/inference/tensorrt/plugins/BUILD]: Add a blank line between targets.

File: modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.cu [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/******************************************************************************
+ * Copyright 2018 The Apollo Authors. All Rights Reserved.
```
Comment (xiaoxq) [inline modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.cu]: 2019

File: modules/perception/inference/tensorrt/plugins/BUILD [hunk 3]
```diff
@@ -11,6 +11,7 @@ cc_library(
         ":perception_inference_tensorrt_plugins_argmax_cuda",
         ":perception_inference_tensorrt_plugins_slice_cuda",
         ":perception_inference_tensorrt_plugins_softmax_cuda",
+        ":perception_inference_tensorrt_plugins_leakyrelu_cuda",
```
Comment (xiaoxq) [inline modules/perception/inference/tensorrt/plugins/BUILD]: Make items sorted.

File: modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.cu [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include <vector>
+#include "modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.h"
```
Comment (xiaoxq) [inline modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.cu]: Put corresponding header file as the first.

File: modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.cu [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace apollo {
+namespace perception {
+namespace inference {
+
+typedef int8_t int8;
```
Comment (xiaoxq) [inline modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.cu]: Unused?

File: modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.h [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    for (int i = 0; i < in_dims.nbDims; i++) {
+      input_dims_.d[i] = in_dims.d[i];
+      input_dims_.type[i] = in_dims.type[i];
+    }
+    negative_slope_ = param.negative_slope();
+  }
+
+  ReLUPlugin() {}
+  ~ReLUPlugin() {}
+  virtual int initialize() { return 0; }
```
Comment (xiaoxq) [inline modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.h]: CamelCase.

Comment (KaWaiTsoiBaidu) [reply modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.h]: you mean `initialize()`? I can't change this name, it is required by the TensorRT library.

File: modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.cu (hunk 4 above)
Comment (KaWaiTsoiBaidu) [reply modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.cu]: Put it before `#include<vector>`?

File: modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.h (hunk 6 above)
Comment (xiaoxq) [reply modules/perception/inference/tensorrt/plugins/leakyReLU_plugin.h]: Got it. Thanks!

**Pull request merged**